routers/web/repo/editor.go

```diff
@@ -41,16 +41,49 @@ const (
 	frmCommitChoiceNewBranch string = "commit-to-new-branch"
 )
 
+func canCreateBasePullRequest(ctx *context.Context) bool {
+	baseRepo := ctx.Repo.Repository.BaseRepo
+	return baseRepo != nil && baseRepo.UnitEnabled(ctx, unit.TypePullRequests)
+}
+
 func renderCommitRights(ctx *context.Context) bool {
 	canCommitToBranch, err := ctx.Repo.CanCommitToBranch(ctx, ctx.Doer)
 	if err != nil {
 		log.Error("CanCommitToBranch: %v", err)
 	}
 	ctx.Data["CanCommitToBranch"] = canCommitToBranch
+	ctx.Data["CanCreatePullRequest"] = ctx.Repo.Repository.UnitEnabled(ctx, unit.TypePullRequests) || canCreateBasePullRequest(ctx)
 
 	return canCommitToBranch.CanCommitToBranch
 }
 
+// redirectForCommitChoice redirects after committing the edit to a branch
+func redirectForCommitChoice(ctx *context.Context, commitChoice, newBranchName, treePath string) {
+	if commitChoice == frmCommitChoiceNewBranch {
+		// Redirect to a pull request when possible
+		redirectToPullRequest := false
+		repo := ctx.Repo.Repository
+		baseBranch := util.PathEscapeSegments(ctx.Repo.BranchName)
+		headBranch := util.PathEscapeSegments(newBranchName)
+		if repo.UnitEnabled(ctx, unit.TypePullRequests) {
+			redirectToPullRequest = true
+		} else if canCreateBasePullRequest(ctx) {
+			redirectToPullRequest = true
+			baseBranch = util.PathEscapeSegments(repo.BaseRepo.DefaultBranch)
+			headBranch = util.PathEscapeSegments(repo.Owner.Name) + "/" + util.PathEscapeSegments(repo.Name) + ":" + headBranch
+			repo = repo.BaseRepo
+		}
+
+		if redirectToPullRequest {
+			ctx.Redirect(repo.Link() + "/compare/" + baseBranch + "..." + headBranch)
+			return
+		}
```

Comment:
Overall LGTM. I think the variables could be escaped when it is used. Otherwise the escaping might be forgotten in the future changes.

And `PathEscapeSegments("a/b:c/d")` outputs `a/b:c/d`

```suggestion
		baseBranch := ctx.Repo.BranchName
		headBranch := newBranchName
		if repo.UnitEnabled(ctx, unit.TypePullRequests) {
			redirectToPullRequest = true
		} else if canCreateBasePullRequest(ctx) {
			redirectToPullRequest = true
			baseBranch = repo.BaseRepo.DefaultBranch
			headBranch = repo.Owner.Name + "/" + repo.Name + ":" + headBranch
			repo = repo.BaseRepo
		}

		if redirectToPullRequest {
			ctx.Redirect(repo.Link() + "/compare/" + util.PathEscapeSegments(baseBranch) + "..." + util.PathEscapeSegments(headBranch))
			return
		}
```